src/common/stattextcmn.cpp

```diff
@@ -213,8 +213,46 @@ class wxLabelWrapper : public wxTextWrapper
 
 void wxStaticTextBase::Wrap(int width)
 {
+    if (width == m_currentWrap) return;
+    m_currentWrap = width;
+
+    // Allow for repeated calls to Wrap with different values
+    if (m_unwrappedLabel.IsNull()) 
+    {
+        m_unwrappedLabel = GetLabel();
+    } 
+    else 
+    {
+        SetLabel( m_unwrappedLabel );
+    }
     wxLabelWrapper wrapper;
     wrapper.WrapLabel(this, width);
+    InvalidateBestSize();
+}
+
+wxSize wxStaticTextBase::GetEffectiveMinSizeFirstPass() const
+{
+    // While wxWrapSizer can only wrap entire controls, a text paragraph
+    // could theoretically wrap at a few letters, so we start with
+    // requesting very little space in the first pass
+    return wxSize( 10, 10 );
```

Comment:
Could we avoid hardcoding magic numbers please? It would make sense to use `GetCharWidth()` or something else based it.